### ISSUE TITLE
Allow disabling refresh on tab focus

### ISSUE THREAD

**Issue by KaKi87** [elk-zone/elk#2484]
Hello,

I often do the following :

- browse a feed ;
- open a post in new background tab ;
- continue browsing feed ;
- focus post tab.

What I expect is : being able to read the post immediately.

What I experience is : needing to wait for the page to refresh while it already had plenty of time to load in the first place.

Please allow disabling this behavior.

Thanks

**Comment by HawtinZeng** [comment]
Sorry, bro, I don't understand what is your problem, can u give me more explanations or offer some video to demonstrate your problem?

**Comment by KaKi87** [comment]
Exactly like I said :

![](https://github.com/elk-zone/elk/assets/21284089/24f9207e-0027-446c-b2af-7296266c78ba)

**Comment by HawtinZeng** [comment]
I known.I will try to fix it

**Comment by HawtinZeng** [comment]
@antfu , there is a user experience flaw, I can't reappear this in my local dev enviroment, but in production env, it acts like Kaki87 said...

**Comment by HawtinZeng** [comment]
I have test it in https://elk.zone/, find that when I ctrl+click an avatar, it will request a manifest-en-US-dark.webmanifest file, then i check to the newly created tab, it request the manifest again
![image](https://github.com/elk-zone/elk/assets/73981739/41d3d4f0-d227-4d39-83c4-7851ac78840e)
@antfu

**Comment by KaKi87** [comment]
Hello,
Any news on this ?
Thanks

**Comment by userquin** [comment]
I'm going to check the page lifecycle, it seems browsers freeze/discard tabs aggresively or we've a bug in the logic.

We need to control those events to close all indexed db connections and control user account switch:
https://developer.chrome.com/docs/web-platform/page-lifecycle-api

**Comment by KaKi87** [comment]
> browsers freeze/discard tabs aggresively 

Nope.

> or we've a bug

Or a feature. But I just want the ability to disable it.

Thanks

**Comment by userquin** [comment]
> > browsers freeze/discard tabs aggresively
> 
> Nope.

From the page lifecycle, **Frozen** state in the table:
> In particular, it's important that you:
>-  Close all open[ IndexedDB](https://developer.mozilla.org/docs/Web/API/IndexedDB_API) connections.
>-  Close open[ BroadcastChannel](https://developer.mozilla.org/docs/Web/API/Broadcast_Channel_API) connections.
>-   Close active WebRTC connections.
>-  Stop any network polling or close any open[ Web Socket](https://developer.mozilla.org/docs/Web/API/WebSockets_API) connections.
>-  Release any held Web Locks.

and some more inputs:

> For the moment, Chrome is going to be conservative when discarding pages and only do so when it's confident it won't affect users. For example, pages that have been observed to do any of the following while in the hidden state will not be discarded unless under extreme resource constraints:
>-   Playing audio
>-    Using WebRTC
>-    Updating the table title or favicon
>-    Showing alerts
>-    Sending push notifications

and from [unload event](https://developer.chrome.com/docs/web-platform/page-lifecycle-api#the_unload_event), maybe we need to include some protection with the `persisted` event prop:
> const terminationEvent = 'onpagehide' in self ? 'pagehide' : 'unload';
window.addEventListener(terminationEvent, (event) => {
  // Note: if the browser is able to cache the page, `event.persisted`
  // is `true`, and the state is frozen rather than terminated.
});

**Comment by userquin** [comment]
> So I went after related listeners and found this : 

The listener is added on  demand and only when the account changes: keeping both tabs in sync (it is inside a watcher).

This issue seems to be about loading the new page content, the problem here is the tab visibility, there are some `useAsyncData` forced to reload the data when page is hydrated, I'm checking if we can fix this.

My previous comments was about switching tabs, from time to time there is a page reload, releted to page lifecycle plugin logic, sorry for the noise.

**Comment by HawtinZeng** [comment]
Maybe the refreshed tab is triggered by this hook.
```javascript
onReactivated(() => {
  // Silently update data when reentering the page
  // The user will see the previous content first, and any changes will be updated to the UI when the request is completed
  refreshStatus()
  refreshContext()
})
```

**Comment by jyn514** [comment]
> The listener is added on demand and only when the account changes: keeping both tabs in sync (it is inside a watcher).

this is not true, it also is added on the first login. i tested and signing out of Elk makes the bug go away (i.e. switching back to the tab becomes instant).

maybe you intended to reload only when the user is changed, not each time the tab gains or loses focus? i can make a PR with that change.

**Comment by userquin** [comment]
There is a bug a bug in 0.15.0 version, should be fixed in main elk.zone, we need a new release to fix the page refresh.

**Comment by jyn514** [comment]
> There is a bug in 0.15.0 version, should be fixed in main elk.zone, we need a new release to fix the page refresh.

hmm, i am also seeing the bug in my local copy of elk.zone built from https://github.com/elk-zone/elk/commit/54344acf4b0834c2bd8e4489bd647ffa14b55fdb.

**Comment by jyn514** [comment]
oh i see, you *just* merged a fix 8 hours ago. yes, with that branch the bug is fixed :) sorry for the noise

**Comment by shuuji3** [comment]
We released v0.15.1 and it should be available at https://elk.zone/. The new version includes some reloading issue fixes by #2975.
